platform/mbed_application.c

```diff
@@ -122,7 +122,7 @@ static void powerdown_scb(uint32_t vtor)
     }
 #endif
     SCB->SHCSR = 0x00000000;
-#if defined(__CORTEX_M23)
+#if defined(__CORTEX_M23) || defined(__CORTEX_M33)
```

Comment:
Cortex-M33 does have these fault status registers and we want to manually clear them, so remove the `defined(__CORTEX_M33)` as the existing code is fine.